trace-dispatcher/doc/trace-dispatcher.md

```diff
@@ -0,0 +1,798 @@
+
+The __logging context__ of the trace is defined as follows:
+
+1. __trace filtering__ -- by __privacy__, __severity__ and __namespace__ context,
+2. __trace presentation__ -- by __detail level__ context.
+
+Severity an detail level can be configured.
+
+## Filter context
+### Severity
+
+__Severity__ is expressed in terms of the enumeration provided by [section 6.2.1 of RFC 5424](https://tools.ietf.org/html/rfc5424#section-6.2.1):
+
+```haskell
+data SeverityS
+    = Debug | Info | Notice | Warning | Error | Critical | Alert | Emergency
+```
+
+..which ranges from minimum (`Debug`) to the maximum (`Emergency`) severity, and allows ignoring messages with severity level _below_ a configured global __severity cutoff__.
+
+The following __trace combinators__ affect __annotated severity__ of a trace:
+
+```haskell
+withSeverity :: Monad m => (a -> Severity) -> Trace m a -> Trace m a
+setSeverity  :: Monad m => Severity        -> Trace m a -> Trace m a
+
+-- Unconditional annotation:
+tracer   = setSeverity Notice trAddBlock
+
+-- Conditional annotation:
+tracer'' = withSeverity (\case
+                           IgnoreBlockOlderThanK{}          -> Warning
+                           IgnoreBlockAlreadyInVolatileDB{} -> Notice)
+                        trAddBlock
+```
+
+If the combinators are applied multiple times to a single trace, only the outermost application affects it -- the rest of them is ignored.
+
+```haskell
+traceWith (setSeverity Warning trAddBlock) (IgnoreBlockOlderThanK b)
+```
+
+In addition, the __severity context__ of a particular trace can be further overriden by configuration, at the __tracer namespace__ granularity -- which allows to put them above or below the __global severity cutoff__, effectively either enabling or disabling them.
+
+`Info` is the default __severity__, in the absence of trace context or configured severity overrides.
+
+NOTE: as en extension to the filtering severity type (`SeverityF`), a `SilenceF` constructor is defined, which encodes unconditional silencing of a particular trace -- and therefore serves as a semantic expression of the `nullTracer` functionality.
+
+### Privacy
+
+__Privacy__ annotations allows limiting __trace-outs__ that particular traces can reach.  It is expressed in terms of:
+
+```haskell
+data Privacy
+    = Confidential | Public
+```
+
+__Confidential__ privacy level means that the trace will not be externalised from the system, except via __standard output__.
+
+The annotation mechanism is similar to the one of severity:
+
+```haskell
+privately :: Trace m a -> Trace m a
+withPrivacy :: Monad m => (a -> Privacy) -> Trace m a -> Trace m a
+```
+
+`Public` is the default __privacy__, in the absence of privacy annotations.
+
+Trace privacy cannot be configured.
+
+See [Confidentiality and privacy filtering implementation](#Confidentiality-and-privacy -filtering-implementation) for a more full discussion of semantics.
```

Comment:
I think you have an accidental space in `Confidentiality-and-privacy -filtering-implementation`